### DISCUSSION TITLE
how to update those installed via `app@version` ??

### DISCUSSION THREAD

**Discussion by goyalyashpal** [ScoopInstaller/Scoop#5954] [Help and Questions] [answered]
#### Current Behavior

* can't [figure out how to] update the app which is 
installed by specifying a version explicitly using `app@version` syntax
* say for example: `myapp@xyz` (older) while the latest available in scoop is `myapp@abc` (newer).

#### Expected Behavior

there exists some _documented_ way to update such apps

#### Additional context/output

[faq]: https://github.com/ScoopInstaller/Scoop/wiki/FAQ#how-do-i-install-a-specific-version-of-an-app "scoop wiki FAQ"

i have:
* read [faq] but can't didn't find any info on how to update that.
* tried everything (listed below) i could think of, but all these show on the lines of app is already update etc.
* one curious detail though, is that both `info`, `cat` use the version from the bucket ("abc"), 
whereas `list` shows this installed version & `<auto-generated>` in the "Source"

```
update, status, cleanup myapp
update myapp
update myapp --force
update extras/myapp --force
install myapp
```

### System details

```powershell
> systeminfo | grep -iE "^os version|system type"
OS Version:                10.0.19045 N/A Build 19045
System Type:               x64-based PC

> powershell -c '& {$PSVersionTable.PSVersion}'

Major  Minor  Build  Revision
-----  -----  -----  --------
5      1      19041  4291

> cat $env:USERPROFILE/.config/scoop/config.json
{
    "root_path":  "D:\\UserFiles\\scoop",
    "global_path":  "D:\\ProgramFiles\\scoop_global",
    "last_update":  "2024-05-....",
    "alias":  {
                  "cat-dum":  "scoop-cat-dum"
              },
    "debug":  true,
    "scoop_branch":  "master",
    "scoop_repo":  "https://github.com/ScoopInstaller/Scoop"
}

> bash -c 'scoop -V 2>&1 | head -n2'
Current Scoop version:
d285bb08 chore(release): Bump to version 0.4.1 (#5924)
```

**Comment by niheaven** [comment]
The only way is uninstall and install again... Since xxx@123 will create a custom manifest that could not be updated.

**Reply by goyalyashpal** [reply]
ouch... i typed a reply then clicked "mark as answer" without posting that - github erased the writeup 😃 

anyhow... retyping:
yep i already did that, all thanks to scoop's persists, caches, install/uninstall scripts which make even hard reinstalls are trivial

also i suspected that the "Source": `<autogenerated>` manifest won't be linked to original manifest

**Reply by niheaven** [reply]
> also i suspected that the "Source": <autogenerated> manifest won't be linked to original manifest

Yes, user wants to mark that version, right?

**Reply by goyalyashpal** [reply]
for my case, i just wanted to install the specified version temporarily for it to match the non-scoop version installed already on the syhtem.

this was so as i thought that it would maybe ease the migration to newer version like ot does in executable installers.

but yeah, scoop updates are just uninstall, install; i was wrong in that idea of version migration

**Comment by goyalyashpal** [comment, marked as answer]
refer to following discussion i created for migrating from custom version of an app to being a normal one:
* https://github.com/ScoopInstaller/Scoop/discussions/5959

summary: update the value of `"bucket": ` field in `$(scoop prefix <app>)/install.json` file to the intended bucket & remove the `"url": ` key-value pair
